### ISSUE TITLE
Unable to select options with apostrophes.

### ISSUE THREAD

**Issue by AndrewDRX** [pytesNET/tail.select#53]
Unable to select options with apostrophes.

Example to reproduce:
```html
<!doctype html>
<html>
  <head>
    <meta charset="utf-8"> 
    <link rel="stylesheet" href="css/tail.select-default.css"></link>
    <script src="js/tail.select.js"></script>
    <script>
      window.onload = function onload() {
        tail.select('#select', {'search': true});
      };
    </script>
  </head>
  <body>
    <div>
      <span>Control:</span>
      <select id="select">
        <option>Default Item</option>
        <option>Number One's Item</option>
        <option>Number Two's Item</option>
        <option>Number Three's Item</option>
      </select>
    </div>
  </body>
</html>
```

**Comment by AndrewDRX** [comment]
Submitted pull reques #54 for this issue.

**Comment by SamBrishes** [comment]
Hellow,

thanks for reporting the bug and fixing them... again! :D 

Sincerely,
Sam.

**Comment by Calak** [comment]
I have to disagree. Look at [this fiddle](https://jsfiddle.net/ra17ejp5/). Searching with a single quote work as expected. (At least, on chrome for android)

**Comment by SamBrishes** [comment]
Hellow,

Thanks for testing too, but the issue concerns the selection of an option (with apostrophes), not the search function. I've tested your jsFiddle, and as shown below I couldn't select the respective option (tested with Mozilla Firefox 67.0.3 on Win10):

![notselectable](https://user-images.githubusercontent.com/31392477/59898071-9c48e300-93ef-11e9-9961-f5c2e267027d.gif)

Sincerely,
Sam.

**Comment by Calak** [comment]
You're right, I misunderstood the issue.

**Comment by Calak** [comment]
Ok then, I investigated a bit more and this bug seems to occurs with strings containing double quotes too. Then, maybe it's safer to also escaping them.
```
    callback: function(item, state, _force){
        var escapeQuote = function (str) { return str.replace(/("|')/g, "\\$1"); }
        var self = this, s = "[data-key='" + escapeQuote(item.key) + "'][data-group='" + escapeQuote(item.group) + "']";

        ....
```

Although it work in this case, if we look deeper at the problem, it looks like that there are much more invalid characters 

> **Note**: Characters that are not part of standard CSS syntax must be escaped using a backslash character. Since JavaScript also uses backslash escaping, be especially careful when writing string literals using these characters. See Escaping special characters for more information.
                                                                                                              from [Document.querySelector()](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector#Notes)


Per example,  [CSS.escape()](https://developer.mozilla.org/fr/docs/Web/API/CSS/escape) or [this polyfill](https://github.com/mathiasbynens/CSS.escape/blob/master/css.escape.js). (You could give a look at [cssesc](https://github.com/mathiasbynens/cssesc) too.)

**Comment by AndrewDRX** [comment]
Hey, @Calak,

I believe that the single quote issue is occurring because the query selector is built using single quotes for the attribute selector, like:
```javascript
"[data-key='" + item.key + "']"
```
So, any additional single quotes in the ```item.key``` value will break the attribute selector.

```
```

Can you post an example where you are able to reproduce with the ```tail.select``` library.

Here is an example using several special characters, where the library is working as desired:
```html
<!doctype html>
<html>
  <head>
    <meta charset="utf-8"> 
    <link rel="stylesheet" href="css/tail.select-default.css"></link>
    <script src="js/tail.select.js"></script>
    <script>
      window.onload = function onload() {
        tail.select('#select', {'search': true});
      };
    </script>
  </head>
  <body>
    <div>
      <span>Control:</span>
      <select id="select">
        <option>Default Item</option>
        <option>"Quoted" Item</option>
        <option>"Solo Quote Item</option>
        <option>Period Mark Item.</option>
        <option>Question Mark Item?</option>
        <option>Exclamation Mark Item!</option>
        <option>Hyphenated-Item!</option>
        <option>(Parenthetical) Item</option>
        <option>[Bracketed] Item</option>
        <option>{Braced} Item</option>
        <option>\ Backslash Item</option>
        <option>\\ Double Backslash Item</option>
        <option>/ Forward slash Item</option>
        <option>// Double Forward Slash Item</option>
        <option>https://github.com/pytesNET/tail.select</option>
        <option>https://github.com/pytesNET/tail.select#hashtag</option>
        <option>https://github.com/pytesNET/tail.select?querystring1=one&querystring2=two</option>
      </select>
    </div>
  </body>
</html>
```

**Comment by Calak** [comment]
Hi @AndrewDRX !

You're right, in fact, we don't have to escaping the CSS selector, but the javascript string itself. 

But there seems to be unhandled cases. The first of these is a key with multiple single quotes. Indeed, ```... .replace("'", "\\'")``` just replaces the first occurrence, not the following ones. 

So, you maybe wonder why don't simply make it a global regex: ``... .replace(/(')/g, "\\'")```.
This brings us to the second problem, with this last regex (and the string-based replacement above) we also add an escaping backslash to single-quotes that are already escaped, leaving them as doubly-escaped single-quotes, and then, the initial problem come again (the backslash itself is escaped, leaving the single-quote unescaped).

Some I came with two solution that seems working:

1. Escaping only single-quotes and their eventually preceding backslashes
```
... .replace(/(\\*)+(')/g, (_, $1, $2) => ($1 ? $1 + $1 : '') + "\\'")
```
2. Escaping all backslashes and all single-quotes
```
... .replace(/('|\\)/g, "\\$1")
```
Obviously, the second one is simpler, shorter and  should be faster. Both seems to work as expected, leaving no edge case unhandled (so far as I know) but maybe need more test cases.



tested with:
```
      <select>
        <option>Default</option>
        <option>"DQ"</option>
        <option>"Solo DQ</option>
        <option>'SQ'</option>
        <option>'Solo SQ</option>
        <option>\"Escaped DQ\"</option>
        <option>Escaped \"solo DQ</option>
        <option>\'Escaped SQ\'</option>
        <option>Escaped \'solo SQ</option>
        <option>\\\"Three BS with a DQ</option>
        <option>Double BS \\' SQ</option>
        <option>Question \noMark Item?</option>
        <option>Exclamation Mark Item!</option>
        <option>Hyphe\nated-Item</option>
        <option>(Parenthetical) Item</option>
        <option>[Bracketed] Item</option>
        <option>{Braced} Item</option>
        <option>\ Backslash Item</option>
        <option>\\ Double Backslash Item</option>
        <option>/ Forward Slash Item</option>
        <option>// Double Forward Slash Item</option>
      </select>
```

**Comment by Calak** [comment]
PR changed to #55

**Comment by AndrewDRX** [comment]
I will note that I noticed an additional issue where backslash characters act as escape characters in the search bar, but that seems like a fix will be unrelated enough that it can probably get its own ticket.

**Comment by SamBrishes** [comment]
Hellow,

This has been fixed / added in Version 0.5.13. Thanks.

Sincerely,
Sam.

**Comment by AndrewDRX** [comment]
@SamBrishes @Calak This issue is not fixed.

**Comment by Calak** [comment]
Yeah the PR #55 was skiped in the last version bump

**Comment by SamBrishes** [comment]
Ooops... Sorry, may fault. 

Added / Fixed in Version 0.5.14.

**Comment by Calak** [comment]
@SamBrishes do you want I double-check that recent PR are included? Or you already did it?
